pkg/dataset/sort_merge_join.go

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dataset
+
+import (
+	"io"
+	"strings"
+)
+
+import (
+	"github.com/pkg/errors"
+)
+
+import (
+	"github.com/arana-db/arana/pkg/mysql/rows"
+	"github.com/arana-db/arana/pkg/proto"
+	"github.com/arana-db/arana/pkg/runtime/ast"
+)
+
+var _ proto.Dataset = (*SortMergeJoin)(nil)
+
+type SortMergeJoin struct {
+	fields     []proto.Field
+	joinColumn *JoinColumn
+	joinType   ast.JoinType
+	outer      proto.Dataset
+	inner      proto.Dataset
+}
+
+type JoinColumn struct {
+	column string
+}
+
+func (j *JoinColumn) Column() string {
+	if j != nil {
+		return j.column
+	}
+
+	return ""
+}
+
+func (s *SortMergeJoin) Close() error {
+	return nil
+}
+
+func (s *SortMergeJoin) Fields() ([]proto.Field, error) {
+	return s.fields, nil
+}
+
+func (s *SortMergeJoin) Next() (proto.Row, error) {
+	// all data is order
```

Comment:
Maybe it's better to move the comment into the SortMergeJoin struct like this:
`
//SortMergeJoin ensures that the data sets participating in the join are in order before using sort-merge join.
type SortMergeJoin struct {
	fields     []proto.Field
	joinColumn *JoinColumn
	joinType   ast.JoinType
	outer      proto.Dataset
	inner      proto.Dataset
}
`